src/vs/editor/contrib/hover/browser/contentHover.ts

```diff
@@ -437,252 +442,385 @@ class ContentHoverVisibleData {
+		this._contentPosition = {
+			position: hoverData.showAtPosition,
+			secondaryPosition: hoverData.showAtSecondaryPosition,
+			positionAffinity: hoverData.isBeforeContent ? PositionAffinity.LeftOfInjectedText : undefined,
+			preference: [preference ?? ContentWidgetPositionPreference.ABOVE]
+		};
+	}
+
+	public showAt(node: DocumentFragment, hoverData: HoverData): void {
+		if (!this._editor || !this._editor.hasModel()) {
+			return;
+		}
+		this._setContentPosition(hoverData);
+		this._render(node, hoverData);
+		const widgetHeight = this._getWidgetHeight();
+		const widgetPosition = hoverData.showAtPosition;
+		this._positionPreference = this._findPositionPreference(widgetHeight, widgetPosition) ?? ContentWidgetPositionPreference.ABOVE;
+		this._setContentPosition(hoverData, this._positionPreference);
 
 		// See https://github.com/microsoft/vscode/issues/140339
 		// TODO: Doing a second layout of the hover after force rendering the editor
 		this.onContentsChanged();
-
-		if (visibleData.stoleFocus) {
-			this._hover.containerDomNode.focus();
+		if (hoverData.stoleFocus) {
+			this._hoverWidget.containerDomNode.focus();
 		}
-		visibleData.colorPicker?.layout();
+		hoverData.colorPicker?.layout();
 	}
 
 	public hide(): void {
-		if (this._visibleData) {
-			const stoleFocus = this._visibleData.stoleFocus;
-			this._setVisibleData(null);
-			this._hoverFocusedKey.set(false);
-			this._editor.layoutContentWidget(this);
-			if (stoleFocus) {
-				this._editor.focus();
-			}
+		if (!this._hoverData) {
+			return;
 		}
-	}
-
-	public onContentsChanged(): void {
+		this._setHoverData(undefined);
+		this._resizableNode.maxSize = new dom.Dimension(Infinity, Infinity);
+		this._resizableNode.clearSashHoverState();
+		this._editor.removeContentWidget(this);
+		this._hoverFocusedKey.set(false);
 		this._editor.layoutContentWidget(this);
-		this._hover.onContentsChanged();
-
-		const scrollDimensions = this._hover.scrollbar.getScrollDimensions();
-		const hasHorizontalScrollbar = (scrollDimensions.scrollWidth > scrollDimensions.width);
-		if (hasHorizontalScrollbar) {
-			// There is just a horizontal scrollbar
-			const extraBottomPadding = `${this._hover.scrollbar.options.horizontalScrollbarSize}px`;
-			if (this._hover.contentsDomNode.style.paddingBottom !== extraBottomPadding) {
-				this._hover.contentsDomNode.style.paddingBottom = extraBottomPadding;
-				this._editor.layoutContentWidget(this);
-				this._hover.onContentsChanged();
-			}
+		if (this._hoverData.stoleFocus) {
+			this._editor.focus();
 		}
 	}
 
-	public clear(): void {
-		this._hover.contentsDomNode.textContent = '';
+	private _setPersistedHoverDimensionsOrRenderNormally(): void {
+		// Added because otherwise the initial size of the hover content is smaller than should be
+		const layoutInfo = this._editor.getLayoutInfo();
+		this._resizableNode.layout(layoutInfo.height, layoutInfo.width);
+		this._setHoverWidgetDimensions('auto', 'auto');
```

Comment:
This confuses me a little because it sets the hover to be really large (the editor size) and then triggers an "auto-layout" to its actual size?

Reply:
I don't recall why I needed this code, I think it was to not make the hover widget size be limited by the resizable node size. I will try without this code.